### DISCUSSION TITLE
Is it possible write a dynamic prompts as a template, use them in other dynamic prompts?

### DISCUSSION THREAD

**Discussion by defaultkavy** [adieyal/sd-dynamic-prompts#325] [Q&A]
I can create the template as these 2 files:
```
# character.txt
{
girl, long hair, black hair,
| boy, short hair, white hair,
}
```

```
# posture.txt
{
running, sportswear,
| walking, school uniform,
| jump,
}
```
I wish I can write like this using wildcard or something else:
```
# prompt input
__character__, __posture__
```
generate:
```
girl, long hair, black hair, walking, school uniform,
```

**Comment by adieyal** [comment]
If you keep it to one line, then you can. 

```
# character.txt
{girl, long hair, black hair,| boy, short hair, white hair,}
```
```
# posture
{running, sportswear,| walking, school uniform,| jump,}
```

then `__character__, __posture__` should work.

I've been wanting to address the single line issue for a while. It would be helpful to be able to build prompts from re-usable components.

**Reply by defaultkavy** [reply]
Looking forward to this feature being implemented soon! I will use the solution you provided first.

**Reply by adieyal** [reply]
@akx  recently overhauled the wildcards mechanism which makes this easier to. implement. It should already be possible with a yaml file, but I haven't tested that yet. I would ideally like to be able to do this using a single text file,defining only one template per file.

In any case, I'll keep you posted.

**Reply by akx** [reply]
Yep, with #341 merged, you can use a YAML file for wildcards and [one of the newline folding styles](https://stackoverflow.com/a/21699210/51685):

```yaml
- >
  {
  girl, long hair, black hair
  | boy, short hair, white hair
  }
- >
  {
  running, sportswear
  | walking, school uniform
  | jump
  }
```

**Comment by scrumpyman** [comment]
Is there any reason to use brackets instead of just putting them on different lines? Just asking in case I'm missing something.

**Reply by adieyal** [reply]
The brackets are needed for the variant. Multiple lines are for readability. This is a pretty simple prompt but I've seen some monster prompts that would take up at least a dozen lines. The idea behind storing individual prompt snippets in  a file means that you can easily start to develop sophisticated prompts from smaller building blocks. 

> Is there any reason to use brackets instead of just putting them on different lines? Just asking in case I'm missing something.

The brackets are needed for the variant. Multiple lines are for readability. This is a pretty simple prompt but I've seen some monster prompts that would take up at least a dozen lines. The idea behind storing individual prompt snippets in  a file means that you can easily start to develop sophisticated prompts from smaller building blocks.
